src/de/mod.rs

```diff
@@ -535,11 +600,34 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Deserializer<'de> {
     }
 
     /// Unsupported. We can’t parse newtypes because we don’t know the underlying type.
-    fn deserialize_newtype_struct<V>(self, _name: &'static str, visitor: V) -> Result<V::Value>
+    fn deserialize_newtype_struct<V>(self, name: &'static str, visitor: V) -> Result<V::Value>
```

Comment:
Not sure why we're touching this in this PR? Maybe this belongs in a separate one? I'm not sure what this is doing. Also, the docs are outdated for this function (they claim newtypes are unsupported).

Reply:
It's for parsing structs which may want to escape strings for one field that's known to be short, but not for another field that might be long.

e.g.

```rust
struct Event<'a> {
    name: heapless::String<16>,
    description: serde_json_core::str::EscapedStr<'a>,
    location_address: serde_json_core::str::EscapedStr<'a>,
}
```

`name` is known to be short, so for convenience it's de-escaped and copied into the `heapless::String`, but `description` and `location_address` might be long, so they borrow from the input.

Reply:
Ah gotcha, that makes sense, but wasn't entirely clear to me upon reviewing `EscapedStr`. Can we add documentation to `EscapedStr` that its supposed to be used for this?